docs/development/go/go-structures/index.md

```diff
@@ -18,235 +19,532 @@ external_resources:
+}
+{{< /file >}}
+
+The output of `pointers.go` will be:
+
+    go run pointers.go
+
+{{< output >}}
+Getting a specific struct field: Nathan
+With implicit dereferencing: Nathan
+FirstName for employeePointer2: Nate
+FirstName for employeePointer1: Nate
+{{< /output >}}
+
+`employeePointer1` points to the memory location of the struct created with the struct literal on line 13. Inserting an ampersand (`&`) before the struct literal (e.g. `Employee{"Nathan", 1201921}`) indicates that the memory location for it should be assigned.
+
+Line 14 shows how to *dereference* the pointer by inserting a `*` before the variable name, which tells Go to return the struct located at the memory location of your pointer. Surrounding this with parentheses and then using dot notation (e.g. `(*employeePointer1).FirstName`) allows you to access fields within the struct.
+
+However, Go allows you to *implicitly* dereference a pointer to a struct in this circumstance. This means that you can simply use normal dot notation (e.g. `employeePointer1.FirstName`) to access fields, even if your struct variable is a pointer.
+
+Lines 17-20 show that creating a second pointer to a struct allows you to manipulate that struct from another variable. In this case, the value of the `FirstName` field for `employeePointer1` has been updated after it was assigned through `employeePointer2` on line 18. This is in contrast with the [value semantics](#value-semantics) demonstrated previously.
+
+### Pointers and Structs and Functions
+
+Passing a pointer to a struct as an argument to a function will allow you to mutate that struct from inside the function scope. Consider this example:
+
+{{< file "pointers.go" go >}}
+package main
+
+import (
+    "fmt"
+)
+
+type Employee struct {
+    FirstName string
+    employeeID int
 }
 
-func processStructure(x myStructure) string {
-    return x.Name + " " + x.Surname
+func ChangeEmployeeID(e *Employee, newID int) {
+    e.employeeID = newID
 }
 
-func processPtr(x *myStructure) string {
-    return (*x).Name + " " + (*x).Surname
+func main() {
+    employeePointer1 := &Employee{"Nathan", 8124011}
+    fmt.Println(*employeePointer1)
+    ChangeEmployeeID(employeePointer1, 1012843)
+    fmt.Println(*employeePointer1)
 }
+{{< /file >}}
+
+The output of `pointers.go` will be:
+
+    go run pointers.go
 
+{{< output >}}
+{Nathan 8124011}
+{Nathan 8124011}
+{{< /output >}}
+
+Alternatively, using this code in the `main` function instead will produce identical results:
+
+{{< file "pointers.go" go >}}
 func main() {
-    ptr := createStruct("Mihalis", "Tsoukalos", 123)
-    s2 := retStructure("Mihalis", "Tsoukalos", 123)
-    fmt.Println((*ptr).Name)
-    fmt.Println(s2.Name)
-    fmt.Println(ptr)
-    fmt.Println(s2)
+    employee1 := Employee{"Nathan", 8124011}
+    fmt.Println(employee1)
+    ChangeEmployeeID(&employee1, 1012843)
+    fmt.Println(employee1)
+}
+{{< /file >}}
+
+## Methods
```

Comment:
New section that gives some explicit details about the method syntax and receiver arguments